### PR TITLE
✨ Rich Text - Add table plugin

### PR DESCRIPTION
**Pull request by Jord-Gui**
Implements #4681 

The rich text editor now natively supports adding a table.
image
**Figure: Three tables created using the rich text editor**

### CODE REVIEW

File: packages/@tinacms/scripts/src/index.ts [hunk 1]
```diff
@@ -30,7 +30,12 @@ interface Option {
 
 const deepMerge = (target, source) => {
   for (const key in source) {
-    if (!source.hasOwnProperty(key) || key === '__proto__' || key === 'constructor') continue;
+    if (
+      !source.hasOwnProperty(key) ||
+      key === '__proto__' ||
+      key === 'constructor'
+    )
+      continue
```
Comment (ncn-ssw) [inline packages/@tinacms/scripts/src/index.ts:38]: **issue**: If `continue` is to broken out to its own line, please use braces as per https://www.ssw.com.au/rules/how-to-avoid-problems-in-if-statements/

Comment (wicksipedia) [reply packages/@tinacms/scripts/src/index.ts:38]: better to have an eslint rules turned on for this
https://eslint.org/docs/latest/rules/brace-style
https://eslint.org/docs/latest/rules/curly

File: packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/index.tsx [hunk 2]
```diff
@@ -57,7 +57,6 @@ export const RichEditor = (props: RichTextType) => {
   )
 
   // This should be a plugin customization
```
Comment (ncn-ssw) [inline packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/index.tsx:59]: **question**: Is this comment referring to the removed line immediately below it, or the 'paragraph' of code?

File: packages/@tinacms/scripts/src/index.ts (hunk 1 above)
Comment (Jord-Gui) [reply packages/@tinacms/scripts/src/index.ts]: I have reverted the change for now as its not relevant to this PR

File: packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/index.tsx (hunk 2 above)
Comment (Jord-Gui) [reply packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/index.tsx]: I have reverted this change as discussed as it is not relevant to this PR

**Review by ncn-ssw — APPROVED** [review]
🦙 Excellent work resolving the many finicky issues that we found in testing

**Pull request merged**